### PR TITLE
Run spellchecker over the code.

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

**Review by ChillFish8 — APPROVED** [review]
LGTM! Nice tool

File: quickwit/quickwit-config/src/quickwit_config/serialize.rs [hunk 1]
```diff
@@ -486,7 +486,7 @@ mod tests {
         .await
         .unwrap_err();
         assert!(format!("{parsing_error:?}")
-            .contains("unknown field `max_num_concurrent_split_searchs`"));
+            .contains("unknown field `max_num_concurrent_split_searches_with_typo`"));
```
Comment (fmassot) [inline quickwit/quickwit-config/src/quickwit_config/serialize.rs:489]: I don't know how this test managed to be successful

Comment (ChillFish8) [reply quickwit/quickwit-config/src/quickwit_config/serialize.rs:489]: The sample is also renamed: https://github.com/quickwit-oss/quickwit/pull/2927/files/6181a93cbd1006b6fe457f10ae789a1f4c48bf4d#diff-cb0bc5c6a7c42d89c3a510f4bb52dbfef2d15f42c3fc478c27aca6b720e560ba

Comment (ChillFish8) [reply quickwit/quickwit-config/src/quickwit_config/serialize.rs:489]: Ah wait damn `searchs` is still spelt wrong

**Pull request merged**